src/ray/pubsub/subscriber.cc

```diff
@@ -95,13 +95,17 @@ SubscriptionCallback SubscriberChannel<KeyIdType>::GetCallbackForPubMessage(
 
   auto maybe_subscription_callback = GetSubscriptionCallback(publisher_address, key_id);
   cum_published_messages_++;
-  if (maybe_subscription_callback.has_value()) {
-    cum_processed_messages_++;
-    // If the object id is still subscribed, return a subscribe callback.
-    return std::move(maybe_subscription_callback.value());
-  } else {
-    return nullptr;
+  if (!maybe_subscription_callback.has_value()) {
+    return;
   }
+  cum_processed_messages_++;
+  // If the object id is still subscribed, run a callback to the callback io service.
+  std::string channel_name =
```

Comment:
not sure, but is name() returning rvalue or lvalue? Should it be const str& ?

Reply:
Let me check it out.